### DISCUSSION TITLE
Return vectors with a search query.

### DISCUSSION THREAD

**Discussion by wammy19** [qdrant/qdrant#64] [Q&A] [answered]
Hi,

Firstly.

I've been running a number of experiments with qdrant and really enjoying it so far, it's pretty magical. :)

Secondly.

Is there a way of returning the vectors as well as the pay load with a search query? I know I can just encode the results again, but I'd like to avoid that computation if I can.

**Comment by wammy19** [comment, marked as answer]
> Hi,
> 
> Firstly.
> 
> I've been running a number of experiments with qdrant and really enjoying it so far, it's pretty magical. :)
> 
> Secondly.
> 
> Is there a way of returning the vectors as well as the pay load with a search query? I know I can just encode the results again, but I'd like to avoid that computation if I can.

I've just noticed you've updated the documentation on your website and found I can retrieve points via id or scroll through the points. It would be cool if you could have the option to return the points with a search query using the python client if that is not yet possible?

**Reply by generall** [reply]
Hi, thanks for your feedback!

Currently you can also use the API https://qdrant.github.io/qdrant/redoc/index.html#operation/get_point which returns a single vector. You can also retrieve multiple vectors in a single request using https://qdrant.tech/documentation/points/#retrieve-points 

In python it would be something like

```python
qdrant_client.http.points_api.get_points(name=collection_name, point_request=PointRequest(ids=ids))
```
You can use found result ids as `ids` in this example.

But I totally agree that having a possibility to get vectors with a search request is a must-have feature. And we even have a related issue https://github.com/qdrant/qdrant/issues/50

**Reply by wammy19** [reply]
Awesome, thanks for this! :D
